From 9d77162ee95df3503a318a911865a9313a711fd7 Mon Sep 17 00:00:00 2001
From: Sean V Kelley <sean.v.kelley@linux.intel.com>
Date: Mon, 3 Dec 2018 09:25:08 -0800
Subject: [PATCH] Install blenderplayer manpage

2019-08-24: Rebased for update to 2.80
2021-02-21: Revised patch subject; removed trailing blank line from
            man page; tabs->spaces for indentation

Signed-off-by: Li, Guangli <guangli.li@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 doc/manpage/blenderplayer.1   | 130 ++++++++++++++++++++++++++++++++++
 source/creator/CMakeLists.txt |   5 ++
 2 files changed, 135 insertions(+)
 create mode 100644 doc/manpage/blenderplayer.1

diff --git a/doc/manpage/blenderplayer.1 b/doc/manpage/blenderplayer.1
new file mode 100644
index 000000000000..57dfb3373e48
--- /dev/null
+++ b/doc/manpage/blenderplayer.1
@@ -0,0 +1,130 @@
+.TH "BLENDERPLAYER" "1" "October 17, 2011" "Blender 2\&.60 (sub 0)"
+.SH "NAME"
+blenderplayer \- the blender game engine runner
+.SH "SYNOPSIS"
+usage:   blenderplayer [\-w [w h l t]] [\-f [fw fh fb ff]] [\-g gamengineoptions] [\-s stereomode] [\-m aasamples] filename.blend
+.SH "DESCRIPTION"
+.B blenderplayer
+is the 3D and physics game engine
+.SH "OPTIONS"
+.PP
+.B -h
+: Prints this command summary
+.PP
+.B -w
+: display in a window
+  optional parameters
+    w = window width
+    h = window height
+    l = window left coordinate
+    t = window top coordinate
+    Note: If w or h is defined, both must be defined.
+    Also, if l or t is defined, all options must be used.
+.PP
+.B -f
+: start game in full screen mode
+  optional parameters
+    fw = full screen mode pixel width
+    fh = full screen mode pixel height
+    fb = full screen mode bits per pixel
+    ff = full screen mode frequency
+    Note: If fw or fh is defined, both must be defined.
+    Also, if fb is used, fw and fh must be used. ff requires all options.
+.PP
+.B -s
+: start player in stereo
+.PP
+    depending on the type of stereo you want:
+.PP
+    stereomode: hwpageflip
+        (Quad buffered shutter glasses)
+.PP
+    syncdoubling
+        (Above Below)
+.PP
+    sidebyside
+        (Left Right)
+.PP
+    anaglyph
+        (Red\-Blue glasses)
+.PP
+    vinterlace
+        (Vertical interlace for autostereo display)
+.PP
+.B -D
+: start player in dome mode
+.PP
+    Optional parameters:
+.PP
+        angle = field of view in degrees
+.PP
+        tilt  = tilt angle in degrees
+.PP
+        warpdata = a file to use for warping the image (absolute path)
+.PP
+        mode = fisheye (Fisheye), truncatedfront (Front\-Truncated),
+               truncatedrear (Rear\-Truncated), cubemap(Cube Map),
+               sphericalpanoramic (Spherical Panoramic)
+.PP
+               depending on the type of dome you are using
+.PP
+.B -m
+: maximum anti-aliasing (eg. 2,4,8,16)
+.PP
+.B -i
+: parent windows ID
+.PP
+.B -d
+: turn debugging on
+.PP
+.B -g
+: game engine options
+.RS 4
+.TP 29
+Name
+Default      Description
+.TP
+---------------------------
+------------------------------------
+.TP
+fixedtime
+0         "Enable all frames"
+.TP
+nomipmap
+0         Disable mipmaps
+.TP
+show_framerate
+0         Show the frame rate
+.TP
+show_properties
+0         Show debug properties
+.TP
+show_profile
+0         Show profiling information
+.TP
+blender_material
+0         Enable material settings
+.TP
+ignore_deprecation_warnings
+1         Ignore deprecation warnings
+.RE
+.PP
+.B -
+: all arguments after this are ignored, allowing python to access them from sys.argv
+.SH "EXAMPLES"
+.TP
+.B blenderplayer -w 320 200 10 10 -g noaudio //home//user//filename.blend
+Launch blenderplayer in window mode with size 320x200 at 10 pixels from left and 10 pixels
+from top of the screen without audio.
+.TP
+.B blenderplayer -g show_framerate = 0 //home//user//filename.blend
+Disable framerate reports.
+.TP
+.B blenderplayer -i 232421 -m 16 //home//user//filename.blend
+Launch blenderplayer embedded in an existing window and set antialiasing to 16.
+
+.SH "SEE ALSO"
+.B blender(1)
+.br
+.SH AUTHORS
+This manpage was written for a Debian by Kevin Roy <kiniou@gmail.com>.
diff --git a/source/creator/CMakeLists.txt b/source/creator/CMakeLists.txt
index 03749f924295..b5c579199cda 100644
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -637,6 +637,7 @@ if(UNIX AND NOT APPLE)
       FILES
         ${CMAKE_SOURCE_DIR}/release/freedesktop/blender.desktop
         ${CMAKE_SOURCE_DIR}/release/freedesktop/icons/scalable/apps/blender.svg
+        ${CMAKE_SOURCE_DIR}/doc/manpage/blenderplayer.1
         ${CMAKE_SOURCE_DIR}/release/freedesktop/icons/symbolic/apps/blender-symbolic.svg
       DESTINATION "."
     )
@@ -1824,6 +1825,10 @@ endif()\n\
           FILES ${CMAKE_CURRENT_BINARY_DIR}/blender.1
           DESTINATION "./share/man/man1"
         )
+      install(
+        FILES ${CMAKE_SOURCE_DIR}/doc/manpage/blenderplayer.1
+        DESTINATION share/man/man1
+      )
       endif()
     endif()
   endif()
-- 
2.41.0
